hashring/rbtree.go

```diff
@@ -234,55 +247,88 @@ func (t *redBlackTree) Delete(val int) bool {
 	return found != nil
 }
 
-func (n *redBlackNode) search(val int) (string, bool) {
-	if n.val == val {
-		return n.str, true
-	} else if val < n.val {
+func (n *redBlackNode) search(key keytype) (valuetype, bool) {
+	cmp := n.key.Compare(key)
+	if cmp == 0 {
+		return n.value, true
+	} else if 0 < cmp {
 		if n.left != nil {
-			return n.left.search(val)
+			return n.left.search(key)
 		}
 	} else if n.right != nil {
-		return n.right.search(val)
+		return n.right.search(key)
+	}
+	return nil, false
```

Comment:
Did you change the return here in purpose? (Used to be `return "", false`). I think `nil` makes more sense but wasn't sure it was intended or an unwanted side-effect